scripts/setup.sh

```diff
@@ -1,14 +1,112 @@
 #!/usr/bin/env bash
-
 set -ex
 
-apt-get update
-source /opt/ros/$ROS_DISTRO/setup.sh
-rosdep update
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --install-ros) ros_distro=$2; shift ;;  
+    esac
+    shift
+done
+
+supported_ros_distros=("melodic")
+
+install_ros(){
+        echo "Installing ROS $ros_distro"
+        #Install ROS Prerequisites
+        apt update
+        apt-get install -y lsb-release gnupg2 curl && apt-get clean all
+        rm -f "/etc/apt/sources.list.d/ros-latest.list"
+        sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list'
+        apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+        curl -sSL 'http://keyserver.ubuntu.com/pks/lookup?op=get&search=0xC1CF6E31E6BADE8868B172B4F42ED6FBAB17C654' | apt-key add - 
+        apt update
+
+        #Install ROS $ros_distro
+        apt install -y ros-$ros_distro-desktop-full
+        source /opt/ros/$ros_distro/setup.bash
+}
+
+
+setup_sample_app(){
+        echo "Setting up sample app"
+        #setup key for colcon bundle
+        apt-key adv --fetch-keys 'http://packages.osrfoundation.org/gazebo.key'
+        apt update
+        apt install -y python-rosdep git
+        if [ ! -f "/etc/ros/rosdep/sources.list.d/20-default.list" ];
+        then
+                rosdep init
+        fi
+        rosdep update
+
+        apt-get install -y python3-apt python3-pip python3-vcstool
+        pip3 install -U setuptools colcon-common-extensions colcon-ros-bundle
+
+        cd robot_ws
+        vcs import < .rosinstall
+        rosdep install --from-paths src --ignore-src -r -y
+        cd ..
+
+        cd simulation_ws
+        vcs import < .rosinstall
+        rosdep install --from-paths src --ignore-src -r -y
+        cd ..
+}
+
+if [ -d "/opt/ros" ];
+then
+        echo "ROS is already installed"
+        if [ ! -z "$ros_distro" ];
+        then
+                echo "ignoring request to install $ros_distro "
```

Comment:
"Ignoring request to install $ros_distro because a ROS installation already exists in your environment."